### PR TITLE
Add quiz combining to the CME package.

### PR DESCRIPTION
**Pull request by charleswaddell**
[PT Story.](https://www.pivotaltracker.com/story/show/81486782) Use the `PcrapFreeCme` database for testing.

### CODE REVIEW

File: CME/pages/CMEEvaluationPage.php [hunk 1]
```diff
@@ -153,6 +255,11 @@ protected function initEvaluation()
 		$this->evaluation = $this->app->getCacheValue($this->getCacheKey());
 
 		if ($this->evaluation === false) {
+			if (!$this->progress->evaluation instanceof CMEQuiz) {
```
Comment (gauthierm) [inline CME/pages/CMEEvaluationPage.php]: Should this check `CMEEvaluation`?

File: CME/pages/CMEQuizPage.php [hunk 2]
```diff
@@ -128,38 +140,124 @@ protected function initInternal()
...
+	// {{{ protected function initProgress()
+
+	protected function initProgress()
+	{
+		$account = $this->app->session->account;
+
+		$progress = $this->getProgress();
+
+		if (!$progress instanceof CMEAccountCMEProgress) {
+			$progress = new CMEAccountCMEProgress();
```
Comment (gauthierm) [inline CME/pages/CMEQuizPage.php]: Should use `SwatDBClassMap`

File: CME/pages/CMEEvaluationPage.php [hunk 3]
```diff
@@ -117,35 +130,124 @@ protected function initInternal()
...
+	// {{{ protected function initProgress()
+
+	protected function initProgress()
+	{
+		$account = $this->app->session->account;
+
+		$progress = $this->getProgress();
+
+		if (!$progress instanceof CMEAccountCMEProgress) {
+			$progress = new CMEAccountCMEProgress();
```
Comment (gauthierm) [inline CME/pages/CMEEvaluationPage.php]: Should use SwatDBClassMap

File: CME/pages/CMEQuizPage.php [hunk 4]
```diff
@@ -392,29 +540,35 @@ protected function saveQuizData(SwatForm $form)
...
+					from AccountEarnedCMECredit
+					where credit = %s and account = %s',
+					$this->app->db->quote($credit->id, 'integer'),
+					$this->app->db->quote($account->id, 'integer')
+				);
 
-				$class_name = SwatDBClassMap::get('CMEAccountEarnedCMECredit');
-				$earned_credit = new $class_name();
-				$earned_credit->setDatabase($this->app->db);
+				if (SwatDB::queryOne($this->app->db, $sql) == 0) {
```
Comment (gauthierm) [inline CME/pages/CMEQuizPage.php]: Triple equals

File: CME/pages/CMEQuizPage.php [hunk 5]
```diff
@@ -209,6 +314,51 @@ protected function initQuiz()
...
+	protected function generateQuiz()
+	{
+		$class_name = SwatDBClassMap::get('CMEQuiz');
+
+		$quiz = new $class_name();
+		$quiz->setDatabase($this->app->db);
+
+		$quiz->createdate = new SwatDate();
+		$quiz->createdate->toUTC();
+		$quiz->save();
```
Comment (gauthierm) [inline CME/pages/CMEQuizPage.php]: set account on quiz before saving

**Comment by gauthierm** [comment]
Quiz response server is not working with the new URL structure.

File: CME/pages/CMEQuizPage.php (hunk 5 above)
Comment (charleswaddell) [inline CME/pages/CMEQuizPage.php]: No need to do this anymore since we have removed the `account` column from Inquisition.

**Comment by charleswaddell** [comment]
> Quiz response server is not working with the new URL structure.

Fixed in cb321357234a011814e9180210fa76374e04fdcd

**Comment by gauthierm** [comment]
`RapCMECreditCondensedMenu` needs to be updated (it wasn't in master when this PR was started)

**Pull request merged**